packages/tailwindcss/src/compat/plugin-api.ts

```diff
@@ -502,9 +502,11 @@ export function objectToAst(rules: CssInJs | CssInJs[]): AstNode[] {
       if (!name.startsWith('--') && value === '@slot') {
         ast.push(rule(name, [atRule('@slot')]))
       } else {
-        // Convert camelCase to kebab-case:
-        // https://github.com/postcss/postcss-js/blob/b3db658b932b42f6ac14ca0b1d50f50c4569805b/parser.js#L30-L35
-        name = name.replace(/([A-Z])/g, '-$1').toLowerCase()
+        if (!name.startsWith('--')) {
```

Comment:
We have this exact same check three lines above, perhaps we can move it outside the `if` into a variable so we only need to do this computation once? 👍

Reply:
Another option is to refactor this a little bit:
```ts
      if (!name.startsWith('--')) {
        if (value === '@slot') {
          ast.push(rule(name, [atRule('@slot')]))
          continue
        }

        // Convert camelCase to kebab-case:
        // https://github.com/postcss/postcss-js/blob/b3db658b932b42f6ac14ca0b1d50f50c4569805b/parser.js#L30-L35
        name = name.replace(/([A-Z])/g, '-$1').toLowerCase()
      }

      ast.push(decl(name, String(value)))
```

Reply:
Went ahead and pushed this so that we can get it merged today 👍